presto-main-base/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -1437,14 +1441,53 @@ else if (character == '%') {
     @SqlType(StandardTypes.INTERVAL_DAY_TO_SECOND)
     public static long parseDuration(@SqlType("varchar(x)") Slice duration)
     {
+        String durationStr = duration.toStringUtf8();
+
+        if (durationStr.isEmpty()) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "duration is empty");
+        }
+
         try {
-            return Duration.valueOf(duration.toStringUtf8()).toMillis();
+            Matcher matcher = PATTERN.matcher(durationStr);
+
+            if (!matcher.matches()) {
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT,
```

Comment:
I'm a bit hesitant about using regex for date/time functions. While convenient, regular expressions can sometimes have performance implications, especially if this code is in a hot path or deals with many date strings. Have we explored alternative, more optimized date parsing methods?